src/query/interpreter.cpp

```diff
@@ -1399,6 +1400,141 @@ PreparedQuery PrepareDumpQuery(ParsedQuery parsed_query, std::map<std::string, T
                        RWType::R};
 }
 
+std::vector<std::vector<TypedValue>> AnalyzeGraphQueryHandler::AnalyzeGraphCreateStatistics(
+    const std::span<std::string> labels, DbAccessor *execution_db_accessor, storage::Storage::Accessor *dba) {
+  auto results = std::vector<std::vector<TypedValue>>();
+  using VertexAccessorIterable = decltype(std::declval<query::DbAccessor>().Vertices(storage::View::OLD));
+  using LPIndex = std::pair<storage::LabelId, storage::PropertyId>;
+  VertexAccessorIterable vertices = execution_db_accessor->Vertices(storage::View::OLD);
+
+  std::map<LPIndex, std::map<storage::PropertyValue, int64_t>> counter;
+
+  // initialize index
+  const auto &indices_info = execution_db_accessor->ListAllIndices();
+  std::for_each(
+      vertices.begin(), vertices.end(), [&indices_info, &counter, &labels, execution_db_accessor](const auto &vertex) {
+        std::for_each(
+            indices_info.label_property.begin(), indices_info.label_property.end(),
+            [&labels, &counter, execution_db_accessor, &vertex](const auto &info) {
+              if (const auto &has_label = vertex.HasLabel(storage::View::OLD, info.first);
+                  has_label.HasError() || !*has_label ||
+                  (labels[0] != "*" && std::find(labels.begin(), labels.end(),
```

Comment:
Is there some other way to write this, so that for example if user wants to analyze on labels `*` that we don't need to check every time HasLabel, but for example do something like this:

```
// preprocess varaible labels so it uses ids already instead of doing LabelToName for every vertex
// let's say we did that and it is stored inside
// labels_set_wanted variable which in case user sent `*` represents all labels inside indices_info in label_property
auto &all_vertex_labels_set = vertex.LabelsSet()

```



std::set_intersection(v1.begin(), v1.end(), v2.begin(), v2.end(),
                          std::back_inserter(v_intersection));

//and afterwards just go over these labels in v_insertion and increment counter

```

You can check here example code:
https://en.cppreference.com/w/cpp/algorithm/set_intersection

Reply:
yep I cleaned it a bit